src/algebra/category/CommRing/adjunctions.lean

```diff
@@ -41,12 +41,19 @@ def free : Type u ⥤ CommRing.{u} :=
 @[simp] lemma free_map_coe {α β : Type u} {f : α → β} :
   ⇑(free.map f) = rename f := rfl
 
+namespace adj
+-- The next two definitions are (unfortunate) "implementation details", helping the elaborator / typeclass search.
+def hom_equiv (X : Type u) (R : CommRing) := hom_equiv R X
+instance (R : CommRing) : is_ring_hom (λ (n : ℤ), (n : R)) := by tidy
+end adj
+open adj
+
 /--
 The free-forgetful adjunction for commutative rings.
 -/
-def adj : free ⊣ forget CommRing :=
+def adj : free ⊣ forget CommRing.{u} :=
```

Comment:
Do we want to have something like the following (probably, under a different name)? I don't propose to do it in this PR.
```lean
class has_free₂ {C D} [concrete_category C] [concrete_category D] [has_forget₂ C D] :=
(free₂ : D ⥤ C)
(adj : free₂ ⊣ forget₂ C D)

abbreviation has_free {C} [concrete_category C] := has_free₂ C (Type u)
```

Reply:
This sounds great! (For a subsequent PR.)